### PR TITLE
BDOG-1338 Update play-auditing and http-verbs dependencies

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,14 +170,6 @@ The following configuration has been renamed, the previous keys are invalid and
 | httpHeadersWhitelist     | bootstrap.http.headersAllowlist    |
 | csrfexceptions.whitelist | bootstrap.csrfexceptions.allowlist |
 
-### Version 0.37.0
-
-#### Play auditing
-play-auditing was bump in major version - changes in [`HttpAuditing#auditFromPlayFrontend`](https://github.com/hmrc/play-auditing/blob/be52847234b113eab713faaeb6b40d00bf4a3bcd/src/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala#L52) signature  were made.
```
Comment (chotai) [inline README.md:176]: This method has been removed as it's not used from anywhere. These outdated comments don't add anything

File: src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala [hunk 2]
```diff
@@ -20,57 +20,25 @@ import org.scalatest.{LoneElement, Matchers, WordSpecLike}
...
-
-    "Return the correct size map when fed with a given amount of items" in {
-      val optionalFields =
-        HeaderFieldsExtractor.optionalAuditFields(Map("Foo" -> "Bar", "Ehh" -> "Meh", "Surrogate" -> "Cool"))
-      optionalFields.loneElement shouldBe ("surrogate" -> "Cool")
-    }
-
-    "Return the correct size map when fed with two identical items" in {
-      val optionalFields = HeaderFieldsExtractor.optionalAuditFields(
-        Map("Foo" -> "Bar", "Ehh" -> "Meh", "Surrogate" -> "Cool", "Surrogate" -> "Cool"))
```
Comment (chotai) [inline src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala:37]: This is just testing that Map works correctly

File: src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala [hunk 3]
```diff
@@ -20,57 +20,25 @@ import org.scalatest.{LoneElement, Matchers, WordSpecLike}
...
-    }
-
-    "Return the correct size map when fed with seq values" in {
-      val optionalFields = HeaderFieldsExtractor.optionalAuditFieldsSeq(
-        Map(
-          "Foo"       -> Seq("Bar"),
-          "Ehh"       -> Seq("Meh"),
-          "Surrogate" -> Seq("Cool"),
-          "Surrogate" -> Seq("Cool", "funk", "grr")))
-      optionalFields.loneElement shouldBe ("surrogate" -> "Cool,funk,grr")
```
Comment (chotai) [inline src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala:48]: This is just testing that Map works correctly

File: src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala [hunk 4]
```diff
@@ -20,57 +20,25 @@ import org.scalatest.{LoneElement, Matchers, WordSpecLike}
 import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.api.test.FakeRequest
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.play.audit.http.HeaderFieldsExtractor
 
 class HttpAuditEventSpec extends WordSpecLike with Matchers with LoneElement with GuiceOneAppPerSuite {
 
-  "The optional audit fields code" should {
```
Comment (chotai) [inline src/test/scala/uk/gov/hmrc/play/bootstrap/config/HttpAuditEventSpec.scala:27]: I've updated and moved these into play-auditing

File: src/main/scala/uk/gov/hmrc/play/bootstrap/config/AuditingConfigProvider.scala [hunk 5]
```diff
@@ -59,10 +61,15 @@ class AuditingConfigProvider @Inject()(
                 .getOrElse(throw new Exception("Missing consumer configuration for auditing"))
             ),
             auditSource = appName,
-            auditExtraHeaders = c.getOptional[Boolean]("auditExtraHeaders")
+            auditSentHeaders = {
```
Comment (colin-lamed) [inline src/main/scala/uk/gov/hmrc/play/bootstrap/config/AuditingConfigProvider.scala]: If this was written:
```scala
auditSentHeaders =
  c.getOptional[Boolean]("auditSentHeaders")
    .orElse(
      c.getOptional[Boolean]("auditExtraHeaders")
        .map(Function.const(_) {
          logger.warn("deprecated config key auditing.auditExtraHeaders - use auditing.auditSentHeaders")
        }
    ).getOrElse(false)
```
the order of precedence would be easier to follow, and the default value would be at the end. Also it wouldn't check `auditExtraHeaders` if `auditSentHeaders` was explicitly set to false.

**Review by colin-lamed — DISMISSED** [review]

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**